File: 武汉地区校赛/2017武大校赛/H-rsp/lib/_CC17.java
import java.util.Scanner;

public class CC17 {
	static private long _input_key = 0,
			    _call_limit = 2000,
				_win_count = 0,
				_last_rand = 0;
	static private final String _accepted = "6d37d07307058f1d44d5dd89cb55060b6c2bf3a1";
	static private final String _wronganswer = "WRONGANSWER";
	static private Scanner scan = null;
	CC17() {
		scan = new Scanner(System.in);
		_last_rand = scan.nextInt();
	}
	int rps(int shape) {
		long result = 0;
		if (_call_limit == 0)
		{
			System.out.println(_wronganswer);
			return 100;
		}
		_call_limit--;
		if (shape < 0 || shape > 2)
		{
			System.out.println(_wronganswer);
			return 100;
		}
		_last_rand = (_last_rand * 65973 + 28411) % 524288;
		result = (shape - _last_rand % 3 + 3) % 3;
		if (result == 2)
		{
			_win_count = 0;
			return -1;
		} else if (result == 0)
		{
			_win_count = 0;
			return 0;
		} else if (result == 1)
		{
			_win_count++;
			if (_win_count == 1000)
			{
				System.out.println(_accepted);
				return 100;
			}
			return 1;
		}
		System.out.println(_wronganswer);
		return 100;
	}
}

File: 武汉地区校赛/2017武大校赛/H-rsp/CC17.py
'''
Package CC17
'''
_initalized = -1
_input_key = 0
_call_limit = 2000
_wincount = 0

_accepted = "6d37d07307058f1d44d5dd89cb55060b6c2bf3a1"
_wronganswer = "WRONGANSWER"

def rps(shape):
    global _initalized, _input_key, _call_limit, _wincount
    result, computer_shape = -1, -1
    if _call_limit == 0:
        print(_accpeted)
        return 100;
    _call_limit -= 1;
    if _initalized == -1:
        _initalized = int(input())
    if (shape < 0) or (shape > 2):
        print(_wronganswer)
        return 100
    _initalized = (_initalized * 65973 + 28411) % 524288
    computer_shape = _initalized % 3
    result = (shape - computer_shape + 3) % 3
    if result == 2:
        _wincount = 0
        return -1
    elif result == 1:
        _wincount += 1
        if _wincount == 1000:
            print(_accepted)
            return 100
        return 1;
    elif result == 0:
        _wincount = 0
        return 0



if __name__ == "__main__":
    print("You should import this package.")
